File: backend/internal/models/watchlist.go
package models

import "time"

type WatchlistFilter struct {
	Page    int
	PerPage int
	UserID  int64
}

type WatchlistItem struct {
	AnimeID     int64     `json:"anime_id"`
	Title       string    `json:"title"`
	Type        string    `json:"type"`
	Status      string    `json:"status"`
	Year        *int16    `json:"year,omitempty"`
	CoverImage  *string   `json:"cover_image,omitempty"`
	MaxEpisodes *int16    `json:"max_episodes,omitempty"`
	AddedAt     time.Time `json:"added_at"`
}

File: frontend/src/types/watchlist.ts
import { AnimeStatus, PaginationMeta } from '@/types/anime'

export interface WatchlistItem {
  anime_id: number
  title: string
  type: string
  status: AnimeStatus
  year?: number
  cover_image?: string
  max_episodes?: number
  added_at: string
}

export interface PaginatedWatchlistResponse {
  data: WatchlistItem[]
  meta: PaginationMeta
}

export interface WatchlistCreateRequest {
  anime_id: number
}

export interface WatchlistCreateResponse {
  data: WatchlistItem
}
